plugins/ocp/ocp-hardware-component-log.c

```diff
@@ -193,16 +196,27 @@ static int get_hwcomp_log_data(struct nvme_dev *dev, struct hwcomp_log *log)
 	}
 #endif /* HWCOMP_DUMMY */
 
+	log_size = le128_to_cpu(log->size);
+
 	print_info("id: %02Xh\n", OCP_LID_HWCOMP);
 	print_info("version: %04Xh\n", log->ver);
 	print_info_array("guid", log->guid, ARRAY_SIZE(log->guid));
-	print_info("size: %s\n", uint128_t_to_string(le128_to_cpu(log->size)));
+	print_info("size: %s\n", uint128_t_to_string(log_size));
+
+	log_bytes = uint128_t_to_double(log_size);
+	if (log->ver == 1)
+		log_bytes *= sizeof(__le32);
+
+	args.len = 0;
+	if (log_bytes > desc_offset)
+		args.len = log_bytes - desc_offset;
+
+	if (!args.len) {
+		print_info_error("error: ocp: invalid hwcomp log size bytes: %.0Lf\n", log_bytes);
+		return -EINVAL;
+	}
```

Comment:
Usually it's better style to test and bail out directly, so

```
if (log_bytes <= desc_offset)
  return err;

args.len = log_bytes - desc_offset;
```

This keeps the logic in order, that is check if valid, then do the operation if it all is good.

Reply:
Yes right so just fixed the changes as mentioned and rebased with the latest master code. Thank you.